outlines/structure.md

```diff
@@ -0,0 +1,56 @@
+# Application structure and code style
+
+1. What's different between Meteor and other kinds of apps?
+    1. All JavaScript, enables code sharing
+    2. Structure is flexible
+    3. Currently everything is about magic file names
+    4. App scope vs. package scope vs. file scope variables
+2. Client vs. server vs. both
+3. Smooth path from prototype to production, three sizes of apps, S, M, L
+4. General stuff
+    1. Filenames - named the same as the export
+        1. Every file exports one thing
+    2. Avoid depending on load order via namespacing and Meteor.startup
+        1. If you depend on load order, your best bet is a package
+    3. Naming guidelines for everything
+        1. https://github.com/meteor/guide/issues/9#issuecomment-148800085
```

Comment:
Yeah, we went through the exact same thought process but it turns out that:

1. You can't always validate on the client. Sometimes you have to validate on the server (eg. uniqueness)
2. `Meteor.error` has a `details` field for a reason. Take a look at what the accounts code does, methods are supposed to throw meaningful errors.
3. It doesn't make sense to both validate on the client and then do the exact same validation in the simulation of the method. You might as well just use the simulation for client side validation.

All of this has led me to very firmly believe that a simpler pattern is possible if the method does the validation. And it is!